### PR TITLE
Feature/map default entity values when mapping in create mode

### PR DESCRIPTION
**Pull request by ConnorStroomberg**
#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] Clean commits
- [ ] No warnings during install
- [ ] Updated flow typing

### CODE REVIEW

File: test/unit/specs/util/test-schemas.js [hunk 1]
```diff
@@ -893,3 +893,98 @@ export const uniqueFieldSchema = {
...
+      'label': 'Username',
+      'labelAttribute': true,
+      'lookupAttribute': true,
+      'maxLength': 255,
+      'name': 'username',
+      'nillable': false,
+      'readOnly': true,
+      'unique': true,
+      'visible': true,
+      'defaultValue': true
```
Comment (fdlk) [inline test/unit/specs/util/test-schemas.js]: defaultValue always contains a string and the client should cast it to the proper type

File: src/util/EntityToFormMapper.js [hunk 2]
```diff
@@ -385,28 +386,39 @@ const generateFormData = (fields: any, data: any, attributes: any) => {
...
         // Recursively generate data for compounds
-        return {...accumulator, ...generateFormData(field.children, data, attribute.attributes)}
+        return {...accumulator, ...generateFormData(field.children, data, attribute.attributes, options)}
       case 'file':
         // Map MOLGENIS FileMeta entity to our form file object
         // which only contains a name
         const fileData = data[field.id]
-        accumulator[field.id] = fileData ? fileData.filename : data[field.id]
+        if (setDefaultValues) {
+          accumulator[field.id] = attribute.defaultValue && attribute.defaultValue.filename
```
Comment (fdlk) [inline src/util/EntityToFormMapper.js]: I'd expect defaultValue to be a string, how can it have a filename attribute?

File: test/e2e/specs/default-values-test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  tags: ['default'], // run this suite with 'yarn e2e --tag default'
+  beforeEach: function (browser) {
+    // Wait for form to be loaded
+    browser.url(browser.globals.devServerURL)
+    browser.url(browser.globals.devServerURL + '/create-entity')
+  },
+
+  'The entity mapper should set the default values in create mode': function (browser) {
+    browser.pause(3000)
+    browser.options.desiredCapabilities.name = 'Default values'
```
[the remark this answers is not in the document]
Comment (ConnorStroomberg) [reply test/e2e/specs/default-values-test.js]: This is some weird hack to get test case name in saucelab logs

File: test/unit/specs/util/test-schemas.js (hunk 1 above)
Comment (ConnorStroomberg) [reply test/unit/specs/util/test-schemas.js]: Why is it designed like this? , I think we should change this in the backend, but will change is for now.

File: src/util/EntityToFormMapper.js (hunk 2 above)
Comment (ConnorStroomberg) [reply src/util/EntityToFormMapper.js]: i expected it would contain a file, but if every thing is a string this is not the case, i will change it.

**Comment by ConnorStroomberg** [comment]
@fdlk "The default values for attribute xdatetime, xcategoricalmref, are not correctly filled in."

The 'xdatetime' value not working seems to be a safari specific issue, and not related to the default value feature ( same issue when loading entity value) .I created a [separate issue for this](https://github.com/molgenis/molgenis-ui-form/issues/211) and suggest not to let it block the PR ( although i think it is important to fix )

**Comment by ConnorStroomberg** [comment]
@fdlk 

Default file values are not working at the moment, i have created a separate issue for this as the fix for this depends on first merging the file detection issue,

[Entity default file values are not correctly mapped to field values](https://github.com/molgenis/molgenis-ui-form/issues/212)

**Pull request merged**